### PR TITLE
fix: remove unneeded cluster index

### LINKED ISSUE #132: Cluster index ${cluster} not needed?
~~~
yq eval ".spokes[${index}].${cluster}.master${master}.mac_ext_dhcp" ${SPOKES_FILE}
vs
yq eval ".spokes[${index}].[].master${master}.mac_ext_dhcp
~~~

### PR DESCRIPTION
**Pull request by iranzo**
# Description

Remove cluster indexes that seem not to be required

Fixes #132

### CODE REVIEW

**Comment by iranzo** [comment]
https://github.com/rh-ecosystem-edge/ztp-pipeline-relocatable/runs/6902700105?check_suite_focus=true

**Review by fbac — APPROVED** [review]
LGTM

**Comment by alknopfler** [comment]
Could I ask why index will not be required ??? just for curiosity...

**Comment by iranzo** [comment]
Sure, check the linked issue where you can see the commands that return exactly the same result

**Comment by alknopfler** [comment]
> Sure, check the linked issue where you can see the commands that return exactly the same result

mmm but what happens if we have 2 or more clusters in the config yaml file? TBH even maybe could be the same result, I don't feel comfortable changing this because this info of cluster could be useful in the future to log for example the name in a easy way....I'd like having this in case we need to refer the cluster name in the future or maybe to make it clear ... 
WDYT???

**Comment by iranzo** [comment]
> > Sure, check the linked issue where you can see the commands that return exactly the same result
> 
> mmm but what happens if we have 2 or more clusters in the config yaml file? TBH even maybe could be the same result, I don't feel comfortable changing this because this info of cluster could be useful in the future to log for example the name in an easy way....I'd like having this in case we need to refer the cluster name in the future or maybe to make it clear ... WDYT???

The 'more than one' cluster is handled via the '${index}' part in the command as it directly branches from the main '.spokes' section, not from within a cluster itself

File: deploy-edgecluster/render_edgeclusters.sh [hunk 1]
```diff
@@ -252,15 +252,15 @@ EOF
     # Now process blocks for each master
     for master in $(echo $(seq 0 $(($(yq eval ".edgeclusters[${edgeclusternumber}].[]|keys" ${EDGECLUSTERS_FILE} | grep master | wc -l) - 1)))); do
         # Master loop
-        export CHANGE_EDGE_MASTER_PUB_INT=$(yq eval ".edgeclusters[${edgeclusternumber}].${cluster}.master${master}.nic_int_static" ${EDGECLUSTERS_FILE})
-        export CHANGE_EDGE_MASTER_MGMT_INT=$(yq eval ".edgeclusters[${edgeclusternumber}].${cluster}.master${master}.nic_ext_dhcp" ${EDGECLUSTERS_FILE})
+        export CHANGE_EDGE_MASTER_PUB_INT=$(yq eval ".edgeclusters[${edgeclusternumber}].[].master${master}.nic_int_static" ${EDGECLUSTERS_FILE})
```
Comment (eifrach) [inline deploy-edgecluster/render_edgeclusters.sh:255]: maybe i have it wrong but i cannot find the list '[]' in the yaml 
```console
$ yq e ".edgeclusters[0].[].master0" examples/config.yaml
Error: Bad expression, please check expression syntax
```
it looks like a dict so this syntx should not work
```shell
yq e ".edgeclusters[0].edgecluster1-name.master0" examples/config.yaml
ignore_ifaces: eno1 eno2
nic_ext_dhcp: eno4 # ext_dhcp -> DHCP
mac_ext_dhcp: "aa:ss:dd:ee:b0:10"
bmc_url: "<url bmc>"
bmc_user: "user-bmc" #TODO  may be clear test and render with script
bmc_pass: "user-pass" #TODO  may be clear test and render with script
root_disk: "/dev/sda"
storage_disk:
  - /dev/sdb
  - /dev/sdc
  - /dev/sde
  - /dev/sdd
  ````

**Review by eifrach — CHANGES_REQUESTED** [review]
hey since this requires a newer version of YQ 
let's bump fist the version on the repo
PR #447

**Comment by iranzo** [comment]
> hey since this requires a newer version of YQ let's bump fist the version on the repo PR #447

Not really sure as it already passed the CI tests with the previous version at https://github.com/rh-ecosystem-edge/ztp-pipeline-relocatable/runs/6902700105?check_suite_focus=true

**Comment by eifrach** [comment]
> > hey since this requires a newer version of YQ let's bump fist the version on the repo PR #447
> 
> Not really sure as it already passed the CI tests with the previous version at https://github.com/rh-ecosystem-edge/ztp-pipeline-relocatable/runs/6902700105?check_suite_focus=true

On the host version it was not updated. The container version is supporting this change

But for good measure we shold bump it

**Comment by eifrach** [comment]
lgtm

**Pull request merged**